### DISCUSSION TITLE
is headless-ui works with twin.micro ?

### DISCUSSION THREAD

**Discussion by Enigma10** [ben-rogerson/twin.macro#445] [Q&A] [answered]
Hi there, 

I am using twin.marco library and we want to replace some of our components with headless-ui. but it is not working properly. 
Has anyone tried headless ui with twin?

**Comment by ben-rogerson** [comment, marked as answer]
Hey Vikash, not sure how I missed your question here but if you're still after some example components, take a look at the new [headless ui examples](https://github.com/ben-rogerson/twin.examples/tree/master/headlessui-typescript) I've added.

**Comment by antoniobrandao** [comment]
@ben-rogerson ahh pity the examples don't include the Combobox (autocomplete). Will try to figure it out tho

**Reply by ben-rogerson** [reply]
Dropdown is very similar - you can use that for inspiration.

**Comment by antoniobrandao** [comment]
@ben-rogerson any chance we get these examples without Typescript one day?

**Reply by ben-rogerson** [reply]
Probably not - but you can convert them to javascript via the command line:

1. `cd` into the `headlessui-typescript` example
2. `npm i @babel/cli @babel/preset-typescript` so we can use newer babel + ts
3. Convert and prettify the files you want to use:
```bash
# Convert
npx babel --no-babelrc --presets @babel/preset-typescript src/components/Disclosure.tsx --out-file src/components/Disclosure.js 
# Prettify
npx prettier src/components/Disclosure.js --write
```

**Reply by antoniobrandao** [reply]
Thanks @ben-rogerson !
